### ISSUE TITLE
Authentication fails for Azure Container Registry

### ISSUE THREAD

**Issue by gavinws** [davedoesdev/dxf#22]
Tested this on versions 5.0.0 and 7.4.0. It seems that Azure has recently changed their response in the Oauth2 workflow. They no longer include a `token` value in the response, only `access_token` is present. Based on my testing the `access_token` value is correct to use; I assume that when `token` was provided it was always the same value as `access_token` (like what Dockerhub does).

```
Traceback (most recent call last):
  File "/Users/gavin/docker_puller/docker_puller.py", line 149, in handler
    dxf.authenticate(username=username, password=password, actions=['pull'])
  File "/Users/gavin/.pyenv/versions/2.7.13/envs/py27/lib/python2.7/site-packages/dxf/__init__.py", line 305, in authenticate
    self.token = r.json()['token']
KeyError: 'token'
```

Seems like `token` is hard-coded as the expected response value:
https://github.com/davedoesdev/dxf/blob/c877b77b7b08912769d6d503b2c955ae4b23742b/dxf/__init__.py#L303-L306

Is it possible to support both values if one is missing?

**Comment by davedoesdev** [comment]
Yes, should be possible. Might be a few days until I get round to it.
Is there an easy way to test against the Azure registry?
Did they write their own registry?

**Comment by davedoesdev** [comment]
https://github.com/Azure/acr/blob/master/docs/AAD-OAuth.md

> The outcome of this operation will be a response with status 200 OK and a body with the following JSON payload:
>
> {"access_token":"eyJ...xcg"}

**Comment by davedoesdev** [comment]
@gavinws please try v7.4.1 and let me know if it works

**Comment by gavinws** [comment]
Looks like pretty much the same thing I did (which indeed works): https://github.com/gavinws/dxf/commit/7e8b36b2e8a1c5f74baf5ea49ed8592d408d6178

:shipit:

**Comment by davedoesdev** [comment]
Implemented in v7.4.1
